Sources/TensorUtils/LogitsWarper/RepetitionPenaltyWarper.swift

```diff
@@ -0,0 +1,25 @@
+import Foundation
+
+/// `RepetitionPenaltyWarper` prevents the repetition of previous tokens through a penalty.
+/// This penalty is applied at most once per token.
+/// https://github.com/huggingface/transformers/blob/main/src/transformers/generation/logits_process.py#L294
+public struct RepetitionPenaltyWarper: LogitsWarper {
+    public var penalty: Float
+
+    public init(penalty: Double) {
+        self.penalty = Float(penalty)
+    }
+
+    public func warp(indices: [Int], logits: [Float]) -> (indices: [Int], logits: [Float]) {
+        var logits = logits
+        for index in indices {
```

Comment:
Did you try to find an accelerated version of this? (not critical right now, but good to keep in mind for the future)

Reply:
I wanted to use gather scatter, but didn't find scatter method in vDSP.